jme3-core/src/main/java/com/jme3/anim/tween/action/Action.java

```diff
@@ -1,22 +1,112 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 package com.jme3.anim.tween.action;
 
 import com.jme3.anim.AnimationMask;
+import com.jme3.anim.Armature;
 import com.jme3.anim.tween.Tween;
 import com.jme3.util.clone.Cloner;
 import com.jme3.util.clone.JmeCloneable;
 
+/**
+ * Base implementation (1st level impl/prototype) of the interface #{@link Tween} for the new animation system.
+ * The Action class encloses an array of {@link Tween} and converts them into jme animation actions if they actually aren't.
+ *
+ * NB : this implementation mimics the {@link com.jme3.anim.tween.AbstractTween}, but it delegates the interpolation method #{@link Tween#interpolate(double)}
+ * to the #{@link BlendableAction} class.
+ *
+ * Subclasses need to make a constructor call passing in the Tween Actions to enclose them & override #{@link Action#interpolate(double)}.
+ *
+ * Example of operation(Uses Abstract Factory Pattern Method(interpolate/doInterpolate)) :
+ *
+ * <li>#Check out #{@link BlendableAction} class.</li>
+ * <li>#Custom Code : </li>
+ * <pre class="prettyprint">
+ *  public abstract class AbstractAction extends Action {
+ *      private double length;
+ *      private volatile float interpolationDelta = 0f;
+ *      public ActionImpl(Tween... tweenActions){
+ *          super(tweenActions);
+ *      }
+ *      public boolean interpolate(double t){
+ *          //animation is out or in a reverse order.
+ *          if(t < 0){
+ *              return true;
+ *          }
+ *          //find where is the interpolation slider for transition.
+ *          //interpolationDelta would be -plugged into the interpolateTransforms(transformI, transformII, delta)- as a shared value.
+ *          interpolationDelta = t / length;
+ *          doInterpolate(t);
+ *          //interpolate as long as the interpolation slider haven't reached the full strength.
+ *          return interpolationDelta < 1.0d;
+ *      }
+ *      public double getLength(){
+ *           return length;
+ *      }
+ *      public void cloneFields(Cloner cloner, Object original) {
+ *         //deep clone fields
+ *      }
+ *      public Action jmeClone() {
+ *          //shallow clone
+ *         try {
+ *             return (Action) super.clone();
+ *         } catch (CloneNotSupportedException exception) {
+ *             throw new RuntimeException(exception);
+ *         }
+ *     }
+ *      //override this factory method to do interpolation
+ *      protected abstract void doInterpolate(double t);
+ *  }
+ * </pre>
+ * <b>Created by Nehon.</b>
+ */
 public abstract class Action implements JmeCloneable, Tween {
 
     protected Action[] actions;
     private double length;
+    //the default speed is 1, which results in a normal tpf.
     private double speed = 1;
     private AnimationMask mask;
     private boolean forward = true;
 
+    /**
+     * Instantiate an action object which encloses a couple of tween animation actions.
```

Comment:
I will change `couple` to `one or more`

i think empty args aren't allowed here and null args as well (i don't know if we should really document that, since most users won't use this class).